plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerConnector.java

```diff
@@ -136,6 +144,14 @@ public List<PropertyMetadata<?>> getTableProperties()
                         "Default limit of rows returned from this table if not specified in the query",
                         null,
                         defaultLimit -> checkProperty(1 <= defaultLimit, INVALID_TABLE_PROPERTY, "default_limit value must be equal or greater than 1"),
+                        false),
+                longProperty(
+                        TableInfo.MAX_DICTIONARY_SIZE,
+                        """
+                                Maximum size of randomly generated dictionaries to pick values from, used for columns with low number of approximate distinct values
+                                observed during table creation using existing data. Set to zero to disable using dictionaries""",
+                        null,
+                        maxDictionarySize -> checkProperty(0 <= maxDictionarySize, INVALID_TABLE_PROPERTY, "max_dictionary_size value must be equal or greater than 0"),
```

Comment:
This property looks weird on table level. Should you rather have per-column property saying if the column is dictionary or not?

Reply:
This allows disabling dictionaries for an individual table. Column properties cannot be set when running a CTAS.